src/pkg/backup/details/details.go

```diff
@@ -682,11 +705,30 @@ func (i ItemInfo) uniqueLocation(baseLoc *path.Builder) (LocationIDer, error) {
 	}
 }
 
+func (i ItemInfo) updateFolder(f *FolderInfo) error {
+	switch {
+	case i.Exchange != nil:
+		return i.Exchange.updateFolder(f)
+
+	case i.OneDrive != nil:
+		return i.OneDrive.updateFolder(f)
+
+	case i.SharePoint != nil:
+		return i.SharePoint.updateFolder(f)
+
+	default:
+		return clues.New("unsupported type")
+	}
+}
+
 type FolderInfo struct {
 	ItemType    ItemType  `json:"itemType,omitempty"`
 	DisplayName string    `json:"displayName"`
 	Modified    time.Time `json:"modified,omitempty"`
 	Size        int64     `json:"size,omitempty"`
+	DataType    ItemType  `json:"dataType,omitempty"`
+	DriveName   string    `json:"driveName,omitempty"`
+	DriveID     string    `json:"driveID,omitempty"`
```

Comment:
This is a bit suspect, since not all folders are children of a drive.  I could see renaming it for something more generic, that expresses container of some kind between the prefix and folders/items, which would need to be interpreted on a case-by-case basis (eg: if sharepoint, the container is a drive).  Or, if more detail is preferred, adding a service subtyping so that folderInfo contains an `ItemInfo` which has the drive ID.

Reply:
>I could see renaming it for something more generic, that expresses container of some kind between the prefix and folders/items

I debated about this with someone else and I'm not real happy with it either. However, it is the most flexible going forward as not populating fields won't cause JSON bloat thanks to `omitempty`. Reusing fields between different data types also doesn't guarantee that we won't need more fields in the future to cover similarly positioned (in the hierarchy) containers. Reusing fields also makes it difficult to determine what they actually mean since the name may not match what it's representing

>adding a service subtyping

service subtyping can be done by examining `DataType`

> folderInfo contains an ItemInfo which has the drive ID

folderInfo is already itself contained within ItemInfo. Additional nesting seems like it would just complicate stuff further

Reply:
> folderInfo is already itself contained within ItemInfo

Ah, my brain had that swapped around, and I was still reading this from the perspective of folder entries, not info.